packages/@tinacms/cli/src/next/commands/build-command/index.ts

```diff
@@ -0,0 +1,330 @@
+import { prodHTML } from './html'
+import { logger, summary } from '../../../logger'
+import { createDBServer, buildProductionSpa } from './server'
+import { Codegen } from '../../codegen'
+import { parseURL } from '@tinacms/schema-tools'
+import {
+  buildASTSchema,
+  buildClientSchema,
+  getIntrospectionQuery,
+} from 'graphql'
+import { diff } from '@graphql-inspector/core'
+import { waitForDB } from './waitForDB'
+
+export class BuildCommand extends Command {
+  static paths = [['build']]
+  rootPath = Option.String('--rootPath', {
+    description:
+      'Specify the root directory to run the CLI from (defaults to current working directory)',
+  })
+  verbose = Option.Boolean('-v, --verbose', false, {
+    description: 'increase verbosity of logged output',
+  })
+  noSDK = Option.Boolean('--noSDK', false, {
+    description: "Don't generate the generated client SDK",
+  })
+  noTelemetry = Option.Boolean('--noTelemetry', false, {
+    description: 'Disable anonymous telemetry that is collected',
+  })
+
+  static usage = Command.Usage({
+    category: `Commands`,
+    description: `Build the CMS and autogenerated modules for usage with Tina Cloud`,
+  })
+
+  async catch(error: any): Promise<void> {
+    // logger.error(error.message)
+    console.log(error)
+    process.exit(1)
+  }
+
+  async execute(): Promise<number | void> {
+    const configManager = new ConfigManager(this.rootPath)
+    logger.info('Starting Tina build')
+
+    try {
+      await configManager.processConfig()
+    } catch (e) {
+      logger.error(e.message)
+      logger.error('Unable to build, please fix your Tina config and try again')
+      process.exit(1)
+    }
+
+    const database = await this.createAndInitializeDatabase(configManager)
+    const { queryDoc, fragDoc } = await buildSchema(
+      database,
+      configManager.config
+    )
+
+    const codegen = new Codegen({
+      schema: await getASTSchema(database),
+      configManager: configManager,
+      noSDK: this.noSDK,
+      queryDoc,
+      fragDoc,
+    })
+    const { apiURL } = await codegen.execute()
+
+    await fs.outputFile(configManager.outputHTMLFilePath, prodHTML)
+
+    await this.checkClientInfo(configManager, apiURL)
+    await this.checkGraphqlSchema(configManager, database, apiURL)
+    await waitForDB(configManager.config, apiURL, false)
+
+    await buildProductionSpa(configManager, database, apiURL)
+
+    // When the build completes, add the gitignore
+    await fs.outputFile(
+      configManager.outputGitignorePath,
+      'index.html\nassets/'
+    )
+
+    summary({
+      heading: 'Tina build complete',
+      items: [
+        {
+          emoji: '🦙',
+          heading: 'Tina Config',
+          subItems: [
+            {
+              key: 'API url',
+              value: apiURL,
+            },
+          ],
+        },
+        {
+          emoji: '🤖',
+          heading: 'Auto-generated files',
+          subItems: [
```

Comment:
Maybe we could add the admin to this list as well? (Seems to be a common place of confusion)